presto-main/src/main/java/com/facebook/presto/server/SimpleHttpResponseHandler.java

```diff
@@ -56,6 +56,9 @@ public void onSuccess(BaseResponse<T> response)
             else if (response.getStatusCode() == HttpStatus.SERVICE_UNAVAILABLE.code()) {
                 callback.failed(new ServiceUnavailableException(uri));
             }
+            else if (response.getStatusCode() == HttpStatus.GONE.code()) {
```

Comment:
when do we get a 401 error back from server. 
This is a client error and usually means that we are trying to access a resource no longer present.